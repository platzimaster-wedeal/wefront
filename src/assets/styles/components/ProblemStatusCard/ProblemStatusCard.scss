@import '../../partials/colors.scss';
@import '../../partials/sizes.scss';
@import '../../Breakpoints.scss';

.problem-status-card {
  box-sizing: border-box;
  position: relative;
  width: 100%;
  display: flex;
  flex-flow: column nowrap;
  padding: $space-size-normal 0;
  margin: $space-size-normal 0;
  cursor: pointer;
  border-top: 1px solid rgba(0,0,0,.1);
  border-bottom: 1px solid rgba(0,0,0,.1);

  &:hover {
    & .problem-status-card__action{
      opacity: 1;
    }

    & .problem-status-card__header h3 {
      color: $color-primary;
    }
  }

  &__header {
    box-sizing: border-box;
    position: relative;
    display: flex;
    flex-flow: row nowrap;
    width: 100%;
    padding: 3px 0;
    border-bottom: 3px solid $color-primary;

    @include media(tablet){
      align-items: flex-end; 
    }

    & h3 {
      margin: 0;
      padding: 0;
      color: $color-black-90;
      transition: .6s ease-in-out;
      @include font-size($font-size-normal);

      @include media(tablet) {
        @include font-size($font-size-medium);
      }
    }
  }

  &__action {
    position: absolute;
    right: 6%;
    top: 42%;
    opacity: 0;
    transition: .6s ease-in-out;

    @include media(tablet) {
      top: 0%;
    }    
  }
}