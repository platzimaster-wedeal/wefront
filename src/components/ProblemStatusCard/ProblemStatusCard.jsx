import React from "react";
import { MdRemoveRedEye } from "react-icons/md";

// Styles
import "../../assets/styles/components/ProblemStatusCard/ProblemStatusCard.scss";

// Components
import ProblemStatus from "../ProblemStatus/ProblemStatus";
import ButtonAction from "../Buttons/ButtonAction";

const ProblemStatusCard = ({
 statusProblem = true,
 idProblem = "#15142",
 nameProblem = "Error with my PC",
}) => {
 return (
  <article className="problem-status-card">
   <div className="problem-status-card__header">
    <h3>{nameProblem}</h3>
    <small>{idProblem}</small>
    <ButtonAction
     icon={MdRemoveRedEye}
     className="problem-status-card__action"
    />
   </div>

   <div className="problem-status-card__status">
    <ProblemStatus status={statusProblem} />
   </div>
  </article>
 );
};

export default ProblemStatusCard;
