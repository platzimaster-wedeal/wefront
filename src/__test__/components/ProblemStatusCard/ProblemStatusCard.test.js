import React from "react";
import { mount } from "enzyme";
import { create } from "react-test-renderer";
import ProblemStatusCard from "../../../components/ProblemStatusCard/ProblemStatusCard";

describe("Test for the component ProblemStatusCard :D", () => {
 const ProblemStatusCardMounted = mount(<ProblemStatusCard />);
 const ProblemStatusCardCreated = create(<ProblemStatusCard />);

 describe("Visual tests for the component", () => {
  test("should the comopnent have the same tree", () => {
   expect(ProblemStatusCardCreated.toJSON()).toMatchSnapshot();
  });
 });

 describe("Functional tests for the component", () => {
  test("Should the component render one element", () => {
   expect(ProblemStatusCardMounted.length).toEqual(1);
  });
  test("Should the title of the component be a string", () => {
   expect(
    typeof ProblemStatusCardMounted.find(
     ".problem-status-card__header h3"
    ).text()
   ).toEqual("string");
  });
  test("Should the title of the component be Worker Information", () => {
   expect(
    typeof ProblemStatusCardMounted.find(
     ".problem-status-card__header small"
    ).text()
   ).toEqual("string");
  });
 });
});
