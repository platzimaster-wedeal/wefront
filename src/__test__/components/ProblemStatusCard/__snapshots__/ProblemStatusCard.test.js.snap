// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the component ProblemStatusCard :D Visual tests for the component should the comopnent have the same tree 1`] = `
<article
  className="problem-status-card"
>
  <div
    className="problem-status-card__header"
  >
    <h3>
      Error with my PC
    </h3>
    <small>
      #15142
    </small>
    <button
      className="button-action problem-status-card__action"
    >
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 4.5C7 4.5 2.73 7.61 1 12c1.73 4.39 6 7.5 11 7.5s9.27-3.11 11-7.5c-1.73-4.39-6-7.5-11-7.5zM12 17c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5zm0-8c-1.66 0-3 1.34-3 3s1.34 3 3 3 3-1.34 3-3-1.34-3-3-3z"
        />
      </svg>
    </button>
  </div>
  <div
    className="problem-status-card__status"
  >
    <div
      className="problem-status problem-status__state--false"
    >
      <svg
        fill="currentColor"
        height="24"
        size="24"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
      <small>
        Solving
      </small>
    </div>
  </div>
</article>
`;
